variable.c

```diff
@@ -1599,59 +1672,49 @@ rb_obj_evacuate_ivs_to_hash_table(ID key, VALUE val, st_data_t arg)
     return ST_CONTINUE;
 }
 
-attr_index_t
-rb_obj_ivar_set(VALUE obj, ID id, VALUE val)
+static VALUE *
+obj_ivar_set_shape_ivar_table(VALUE obj, void *_data)
 {
-    attr_index_t index;
-
-    rb_shape_t *shape = rb_shape_get_shape(obj);
-    uint32_t num_iv = shape->capacity;
-
-    if (rb_shape_obj_too_complex(obj)) {
-        rb_complex_ivar_set(obj, id, val);
-        return 0;
-    }
-
-    rb_shape_t *next_shape;
-
-    if (!rb_shape_get_iv_index(shape, id, &index)) {
-        index = shape->next_iv_index;
-        if (index >= MAX_IVARS) {
-            rb_raise(rb_eArgError, "too many instance variables");
-        }
+    RUBY_ASSERT(!rb_shape_obj_too_complex(obj));
 
-        RUBY_ASSERT(!rb_shape_obj_too_complex(obj));
+    return ROBJECT_IVPTR(obj);
+}
 
-        if (UNLIKELY(shape->next_iv_index >= num_iv)) {
-            RUBY_ASSERT(shape->next_iv_index == num_iv);
+static void
+obj_ivar_set_shape_resize_table(VALUE obj, attr_index_t old_capa, attr_index_t new_capa, void *_data)
```

Comment:
nitpick: the `_table` suffix make me think of `st_table`, as in TOO_COMPLEX.

Reply:
I renamed it `ivptr` 👍